### ISSUE TITLE
Support for Serial on Particle devices

### ISSUE THREAD

**Issue by pkourany** [rwaldron/spark-io#60]
VoodooSpark supports serial data functions but spark-io does not.  Will you be adding this capability soon or are there specific barriers to doing so?

**Comment by rwaldron** [comment]
> VoodooSpark supports serial data functions

These are only used for debugging when the spark is connected via USB.

> spark-io does not. Will you be adding this capability soon or are there specific barriers to doing so?

I'm not sure what capability you're referring to? Do you want to be able to control the Spark core over USB? If that's the case, then you don't need to use an IO Plugin with Johnny-Five, just load [firmata/spark](https://github.com/firmata/spark) onto the Core and connect as you would with an Arduino family board.

**Comment by pkourany** [comment]
I am looking to be able to read/write on Serial1 through voodoospark (from node.js server) but spark-io does not support that I believe.

**Comment by rwaldron** [comment]
> I am looking to be able to read/write on Serial1 through voodoospark (from node.js server) but spark-io does not support that I believe.

That's correct. Spark-IO implements an emerging standard for [IO-Plugins](https://github.com/rwaldron/io-plugins/), which doesn't currently specify a way to interact with Serial, because in many cases Serial is being used to communicate with the actual board. Any API specification added to IO Plugins must be defined in a way that will work for all of the platforms that are currently supported and hopefully for any new ones in the future. I'd be willing to implement some experimental features to get the ball rolling towards support for serial read/write. I'm at JSConf this week, but I'll start roughing out some ideas and report back as soon as I have something worth test driving—hopefull by Monday?
